### DISCUSSION TITLE
How to copy File to clipboard?

### DISCUSSION THREAD

**Discussion by DmitryBorodiy** [AvaloniaUI/Avalonia#15557] [General]
Hello, Avalonia community!
I need to copy file from storage to clipboard, how I can do it with Avalonia?

Thank you in advance!

**Comment by maxkatz6** [comment]
Simpler but Desktop specific API:
```
var obj = new DataObject();
obj.Set(DataFormats.FileNames, "C://FileName.txt");
await window.Clipboard.SetDataObjectAsync(obj);
```

A bit more recommended API:
```
// Assuming that you need to get files from somewhere:
var files = await window.StorageProvider.OpenFilePickerAsync(new FilePickerOpenOptions());

var obj = new DataObject();
obj.Set(DataFormats.Files, files);
await window.Clipboard.SetDataObjectAsync(obj);
```

**Reply by maxkatz6** [reply]
Although, it seems SetDataObjectAsync isn't implemented on mobile right now, so either API is fine.

**Reply by DmitryBorodiy** [reply]
Okay, I try it. Thank you a kind person!

**Reply by DmitryBorodiy** [reply]
I try this but this is not work for me!

**Comment by DmitryBorodiy** [comment]
I try this but this is not works for me!

**Comment by DmitryBorodiy** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/57910623/4c800a75-add8-4ebb-94e2-17d3512fc803)

CanDo is true now. And operation is running without any exception but file is not on clipboard.

**Reply by stevemonaco** [reply]
Please don't use `async void` for anything except an event handler. Your methods here should be `async Task` instead. Otherwise, any exceptions might be swallowed. It probably isn't your issue though.

**Comment by Koichi-hub** [comment]
Hello stranger! This discussion was closed, but i have solution for you
Over here:
```c#
if (Application.Current?.ApplicationLifetime is not IClassicDesktopStyleApplicationLifetime desktop ||
    desktop.MainWindow?.Clipboard is not { } clipboard || desktop.MainWindow?.StorageProvider is not { } storageProvider)
{
    return;
}

IStorageFile? file = await storageProvider.TryGetFileFromPathAsync("/some/path/to/file");
if (file != null)
{
    DataObject dataObject = new();
    dataObject.Set(DataFormats.Files, new List<IStorageFile> { file });
    await clipboard.SetDataObjectAsync(dataObject);
}
```

It was worked for me. I found this line `storageProvider.TryGetFileFromPathAsync` [here](https://github.com/AvaloniaUI/Avalonia/blob/b16975cb0402cd10358f17135d25c3dd3aac7ac6/samples/ControlCatalog/Pages/ClipboardPage.xaml.cs#L90)

This code let you copy file by path in clipboard
